apps/site/assets/ts/schedule/components/schedule-finder/ServiceSelector.tsx

```diff
@@ -194,7 +178,8 @@ export const ServiceSelector = ({
               route: routeId,
               origin: stopId,
               direction: directionId,
-              date: selectedService.end_date
+              date: services.find(service => service.id === selectedServiceId)!
```

Comment:
Can we avoid the need to not-null this? For example, pick the first service, in the event that there's no selected service? Or do we know for sure this cannot ever be null, and need to get the compiler to understand that?

Reply:
Sadly this is a latter case, as Array.find is typed to possibly return undefined

Reply:
Yes, now I have -- see latest commit. We still need to use the ID for the select onChange event.